thunder/executors/torchex.py

```diff
@@ -1668,6 +1672,36 @@ def max_pool3d_bwd_wrapper(
 ex.register_implementation(
     ltorch.max_pool3d, max_pool3d, checker=_always_executable, grad_transform=max_pool3d_bwd_wrapper
 )
+
+
+def adaptive_avg_pool2d_bwd_wrapper(
+    a: TensorProxy,
+    /,
+    output_size: int | Sequence[int],
+) -> TensorProxy:
+    primals = adaptive_avg_pool2d(a, output_size)
+
+    grad = get_grad(primals)
+    grad_a = adaptive_avg_pool2d_backward(grad, a)
+    put_grad(a, grad_a)
+
+    return primals
+
+
+ex.register_implementation(
+    ltorch.adaptive_avg_pool2d,
+    adaptive_avg_pool2d,
+    checker=_always_executable,
+    grad_transform=adaptive_avg_pool2d_bwd_wrapper,
+)
+_register_implementation(ltorch.adaptive_avg_pool2d_backward, adaptive_avg_pool2d_backward, checker=_always_executable)
+ex.register_implementation(
+    prims.adaptive_avg_pool2d,
+    adaptive_avg_pool2d,
+    checker=_always_executable,
+    grad_transform=adaptive_avg_pool2d_bwd_wrapper,
+)
+_register_implementation(prims.adaptive_avg_pool2d_backward, adaptive_avg_pool2d_backward, checker=_always_executable)
```

Comment:
If the meta functions implementation is moved from `prims.py` to `thunder/torch/__init__.py` then we wouldn't need this code.
```suggestion
```